src/graphnet/data/i3extractor.py

```diff
@@ -327,36 +327,89 @@ def __call__(self, frame, padding_value=-1) -> dict:
             "SubrunID": frame["I3EventHeader"].sub_run_id,
             "EventID": frame["I3EventHeader"].event_id,
             "SubEventID": frame["I3EventHeader"].sub_event_id,
+            "dbang_decay_length": padding_value,
+            "track_length": padding_value,
+            "stopped_muon": padding_value,
             "DeepCoreFilter_13": padding_value,
             "CascadeFilter_13": padding_value,
             "MuonFilter_13": padding_value,
             "OnlineL2Filter_17": padding_value,
-            "dbang_decay_length": padding_value,
-            "track_length": padding_value,
-            "stopped_muon": padding_value,
+            "L3_oscNext_bool": padding_value,
+            "L4_oscNext_bool": padding_value,
+            "L5_oscNext_bool": padding_value,
+            "L6_oscNext_bool": padding_value,
+            "L7_oscNext_bool": padding_value,
         }
 
         if (
             frame["I3EventHeader"].sub_event_stream == "InIceSplit"
         ):  # only inicesplit p frames have filters calculated
-            output.update(
-                {
-                    "DeepCoreFilter_13": int(
-                        bool(frame["FilterMask"]["DeepCoreFilter_13"])
-                    ),
-                    "CascadeFilter_13": int(
-                        bool(frame["FilterMask"]["CascadeFilter_13"])
-                    ),
-                    "MuonFilter_13": int(
-                        bool(frame["FilterMask"]["MuonFilter_13"])
-                    ),
-                    "OnlineL2Filter_17": int(
-                        bool(frame["FilterMask"]["OnlineL2Filter_17"])
-                    ),
-                }
-            )
 
-        if is_mc and (not is_noise):
+            if frame_has_key(frame, key="FilterMask"):
+                output["DeepCoreFilter_13"] = turn_bool_into_number(
+                    try_get_key(
+                        frame["FilterMask"],
+                        "DeepCoreFilter_13",
+                        default_value=-1,
+                    )
+                )  # what if frame doesnt have deepcorefilter and it says true (bool(-1)=true)
```

Comment:
I'll be suggesting these sorts of changes throughout as (1) I think they're easier to read and understand than the currently proposed code and (2) they don't rely on somewhat opaque methods such as `turn_bool_into_number`. It is also using the fact that a default value (`padding_value`) has already been set above, so if they key doesn't exist we don't _need_ to assign a default value here.
```suggestion
                if "DeepCoreFilter_13" in frame["FilterMask"]:
                    output["DeepCoreFilter_13"] = int(frame["FilterMask"]["DeepCoreFilter_13"])
```